core/src/main/scala/kafka/server/DynamicBrokerConfig.scala

```diff
@@ -677,13 +677,15 @@ class DynamicLogConfig(logManager: LogManager, server: KafkaBroker) extends Brok
     val currentLogConfig = logManager.currentDefaultConfig
     val origUncleanLeaderElectionEnable = logManager.currentDefaultConfig.uncleanLeaderElectionEnable
     val newBrokerDefaults = new util.HashMap[String, Object](currentLogConfig.originals)
-    newConfig.valuesFromThisConfig.forEach { (k, v) =>
-      if (DynamicLogConfig.ReconfigurableConfigs.contains(k)) {
-        DynamicLogConfig.KafkaConfigToLogConfigName.get(k).foreach { configName =>
-          if (v == null)
-             newBrokerDefaults.remove(configName)
-          else
-            newBrokerDefaults.put(configName, v.asInstanceOf[AnyRef])
+    // Call extractLogConfigMap to get the new LogConfig because there are some special handling for some configs, e.g.,
+    // RetentionMsProp. This is to ensure that the new LogConfig is consistent with LogManager's behavior.
```

Comment:
Please extend the comment to describe the rationale behind using TopicConfigSynonyms, too.

Reply:
Added more comment.